modules/deployment-model/src/test/java/org/mule/runtime/deployment/model/internal/plugin/deployment/AbstractClassloaderModelLoaderTestCase.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.deployment.model.internal.plugin.deployment;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.endsWith;
+import static org.hamcrest.Matchers.is;
+import org.mule.runtime.deployment.model.api.plugin.loader.MalformedPluginException;
+import org.mule.runtime.deployment.model.api.plugin.loader.Plugin;
+import org.mule.runtime.deployment.model.api.plugin.PluginDescriptor;
+import org.mule.runtime.deployment.model.api.plugin.dependency.ArtifactDependency;
+import org.mule.runtime.deployment.model.api.plugin.classloadermodel.ClassloaderModel;
+import org.mule.runtime.deployment.model.api.plugin.classloadermodel.MalformedClassloaderModelException;
+import org.mule.runtime.deployment.model.internal.plugin.classloadermodel.ClassloaderModelLoader;
+import org.mule.runtime.deployment.model.internal.plugin.dependency.DefaultArtifactDependency;
+import org.mule.tck.junit4.AbstractMuleTestCase;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public abstract class AbstractClassloaderModelLoaderTestCase extends AbstractMuleTestCase {
+
+  protected abstract Plugin getPlugin(String pluginParentFolder, String pluginFolder) throws MalformedPluginException;
+
+  protected abstract String getRuntimeClasses();
+
+  @Parameterized.Parameter(value = 0)
+  public String pluginParentFolder;
+  /**
+   * as plugin.properties do NOT have the plugin's name, we need to know whether we are testing against a properties or
+   * a json classloadermodel file to generate a proper plugin's expected name in {@link #assertMinimalRequirementsPlugin(PluginDescriptor, String)}
```

Comment:
This class is similar to SuccessfulAbstractPluginTestCase. Same comment form that class apply here